### PR TITLE
Ticket #1: Added functions findNext / PreviousInDateOrder

### PR DESCRIPTION
**Pull request by FabianLiebl**

### CODE REVIEW

File: src/Enhavo/Bundle/ArticleBundle/Repository/ArticleRepository.php [hunk 1]
```diff
@@ -84,4 +84,79 @@ public function findPublished()
...
+        $query->setMaxResults(1);
+        $nextArticle = $query->getQuery()->getResult();
+        if (empty($nextArticle)) {
+            // No newer articles
+            return null;
+        } else {
+            return $nextArticle[0];
+        }
+    }
+    public function findPreviousInDateOrder(Article $currentArticle)
```
Comment (gseidel) [inline src/Enhavo/Bundle/ArticleBundle/Repository/ArticleRepository.php]: missing empty line and doc block

File: src/Enhavo/Bundle/ArticleBundle/Repository/ArticleRepository.php [hunk 2]
```diff
@@ -84,4 +84,79 @@ public function findPublished()
 
         return $articles;
     }
+
+    public function findNextInDateOrder(Article $currentArticle)
```
Comment (gseidel) [inline src/Enhavo/Bundle/ArticleBundle/Repository/ArticleRepository.php]: missing doc block

File: src/Enhavo/Bundle/ArticleBundle/Repository/ArticleRepository.php [hunk 3]
```diff
@@ -84,4 +84,79 @@ public function findPublished()
...
+    {
+        // Find articles with same date
+        $query = $this->createQueryBuilder('n');
+        $query->andWhere('n.public = true');
+        $query->andWhere('n.publication_date <= :currentDate');
+        $query->andWhere('n.publication_date = :articleDate');
+        $query->setParameter('currentDate', new \DateTime());
```
Comment (gseidel) [inline src/Enhavo/Bundle/ArticleBundle/Repository/ArticleRepository.php]: please give option the set the current date via parameter

File: src/Enhavo/Bundle/ArticleBundle/Repository/ArticleRepository.php [hunk 4]
```diff
@@ -84,4 +84,79 @@ public function findPublished()
...
+                }
+            }
+        }
+
+        // No next one at current date, search for newer ones
+        $query = $this->createQueryBuilder('n');
+        $query->andWhere('n.public = true');
+        $query->andWhere('n.publication_date <= :currentDate');
+        $query->andWhere('n.publication_date < :articleDate');
+        $query->setParameter('currentDate', new \DateTime());
```
Comment (gseidel) [inline src/Enhavo/Bundle/ArticleBundle/Repository/ArticleRepository.php]: please give option the set the current date via parameter

File: src/Enhavo/Bundle/ArticleBundle/Repository/ArticleRepository.php (hunk 3 above)
Comment (FabianLiebl) [inline src/Enhavo/Bundle/ArticleBundle/Repository/ArticleRepository.php]: The filters for "public=true" and "publication date<=current date" are there for the list to be consistent with the other functions (findPublished, filterByDate, getMonths). These all have the same filters, and none of them can set the current date parameter. I recommend keeping it that way or changing it for all of these functions.

**Pull request merged**